src/components/limited-restoration-panel/LimitedRestorationPanel.vue

```diff
@@ -1,127 +1,147 @@
+      name="radioValue"
+      label="2 years"
+      value="24"
+    />
+    <v-radio
+      id="radio-18"
+      class="radio-button pt-2"
+      name="radioValue"
+      label="18 months"
+      value="18"
+    />
+    <v-radio
+      id="radio-12"
+      class="radio-button pt-2"
+      name="radioValue"
+      label="12 months"
+      value="12"
+    />
+    <v-radio
+      id="radio-6"
+      class="radio-button pt-2"
+      name="radioValue"
+      label="6 months"
+      value="6"
+    />
     <v-row class="ml-0 mt-0 radio-button pt-2">
-      <v-radio class="mt-n4" id="custom-months" name="selectMonths" value="customMonths" />
+      <v-radio
+        id="radio-custom"
+        class="mt-n4"
+        name="radioValue"
+        value="customMonths"
+      />
       <v-form ref="monthsRef">
         <v-text-field
-          id="months-text-field"
-          class="number-months-field"
+          id="text-field-months"
+          v-model="inputValue"
+          class="text-field-months"
           type="number"
           dense
+          filled
           hide-spin-buttons
-          :rules="monthRules"
-          v-model="numberOfMonths"
-          :disabled="!customMonths"
-          @change="onMonthsChanged"
-          filled />
+          :rules="monthsRules"
+          :disabled="(radioValue !== 'customMonths')"
+          @input="onMonthsInput($event)"
+        />
       </v-form>
-      <div class="ml-2 mt-2 month-text">month(s)</div>
+      <span class="ml-2 mt-2 month-text">month(s)</span>
     </v-row>
   </v-radio-group>
 </template>
 
 <script lang="ts">
-import { Component, Emit, Mixins, Prop, Watch } from 'vue-property-decorator'
-import { DateMixin } from '@/mixins' // NB: local mixin (StoryBook can't find it otherwise)
+import Vue from 'vue'
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator'
 import { FormIF } from '@bcrs-shared-components/interfaces'
 
 @Component({})
-export default class LimitedRestorationPanel extends Mixins(DateMixin) {
+export default class LimitedRestorationPanel extends Vue {
   // Refs
   $refs!: Vue['$refs'] & {
     monthsRef: FormIF
   }
 
-  @Prop({ default: '' }) readonly currentDate!: string
-
-  @Prop({ default: '' }) readonly expiryDate!: string
-
+  @Prop({ required: true }) readonly months!: number
   @Prop({ default: 24 }) readonly maxNumberOfMonths!: number
 
   // Local properties
-  private customMonths = ''
-  private selectMonths = ''
-  private numberOfMonths: number = null
-  private monthRules: ((data: string) => void)[] = []
+  radioValue = '' // v-model for radio buttons
+  inputValue = '' // v-model for months text field
+
+  /**
+   * The validation rules for months input.
+   */
+  get monthsRules (): Array<(v: string) => boolean | string> {
+    return [
+      (v) => (+v > 0 && +v <= this.maxNumberOfMonths) || 'Must be between 1 and ' + this.maxNumberOfMonths,
+      (v) => (+v % 1 === 0) || 'Must be a whole number'
+    ]
+  }
```

Comment:
This was previously set from the `mounted()` lifecycle event, and only ever had one value, so I moved it here. Note that it's a getter (computed) instead of a static array because it uses a variable internally.